### DISCUSSION TITLE
vips_image_get_filename not returning the real file the image was loaded from

### DISCUSSION THREAD

**Discussion by johntrunc** [libvips/libvips#2255] [Q&A] [answered]
Hello,

I'm using the function `vips_image_get_filename` to get the file the image was loaded from, working fine for all the format files except with GIF files.

With the following C code:

```c
// gcc -g -Wall example.c `pkg-config vips --cflags --libs`
#include <stdio.h>
#include <vips/vips.h>

int
main(int argc, char **argv)
{
  if (VIPS_INIT("getfilename")) {
    vips_error_exit(NULL);
  }

  VipsImage *jpeg_image, *gif_image;

  if (!(jpeg_image = vips_image_new_from_file("example.jpeg", NULL))) {
    vips_error_exit(NULL);
  }

  printf("filename (jpeg): %s\n", vips_image_get_filename(jpeg_image));

  if (!(gif_image = vips_image_new_from_file("example.gif", NULL))) {
    vips_error_exit(NULL);
  }

  printf("filename (gif): %s\n", vips_image_get_filename(gif_image));

  return(0);
}
```

I'm getting the following output:

```
filename (jpeg): example.jpeg
filename (gif): temp-1
```

For the GIF file I expect to return `example.gif` instead of `temp-1`. Is something not working as expected or has the gif loader a different behavior?.

Thank you.

**Comment by jcupitt** [comment, marked as answer]
Yes, that's a minor bug in the gif loader. It's fixed in master:

```
john@banana ~/pics $ vipsheader -f filename 3198.gif 
"3198.gif"
```

**Reply by johntrunc** [reply]
I see. I will try to investigate a possible workaround or wait until the next release. 

As always thanks for your help @jcupitt
